client/src/legacy/SudoModePasswordField/SudoModePasswordFieldEntwine.js

```diff
@@ -0,0 +1,63 @@
+/* global window */
+import jQuery from 'jquery';
+import React from 'react';
+import { createRoot } from 'react-dom/client';
+import { loadComponent } from 'lib/Injector';
+import qs from 'qs';
+
+jQuery.entwine('ss', ($) => {
+  $('.js-injector-boot .SudoModePasswordField').entwine({
+    Component: null,
+    ReactRoot: null,
+
+    onmatch() {
+      // onmatch will match both the field holder and the field
+      // we only want to run this on the field holder
+      if (this.is('input')) {
+        return;
+      }
+      this._super();
+      const cmsContent = this.closest('.cms-content').attr('id');
+      const context = (cmsContent)
+        ? { context: cmsContent }
+        : {};
+      const SudoModePasswordField = loadComponent('SudoModePasswordField', context);
+      this.setComponent(SudoModePasswordField);
+      this.refresh();
```

Comment:
Why do we refresh on match? What other scenarios call refresh on this component?

Reply:
refresh() is a method defined in this file. I based this file I think from LinkField.js which has the same thing.

Reply:
If you're only intending for it to be called on match, just move the code straight into the onmatch event handler.
This makes it look like `refresh()` will be called from other places in some scenarios.

Reply:
I think it means the component will rerender in some scenarios, like if the large fragment of html gets pjaxed in. Maybe?

It's following an existing pattern will use for other things e.g. LinkField. I don't know why we have that pattern, maybe it does nothing, though I see no downside in just following it for better standardization.

Reply:
That pattern is in other places explicitly because `refresh()` is intended to be called from various scenarios on some components.

It's probably in linkfield erroneously, and causes confusion having this method which is used elsewhere for specific purposes but is here just to be here, because it was seen somewhere else. This method is causing confusion and from what I can tell actually breaking standardisation by making it unclear when the `refresh()` method is serving a purpose and when it's not.

Reply:
:shrug: Honestly it's doing no harm so let's just keep as is, I don't think we need to go back and forth over this

Reply:
I think it is doing harm, as I've already mentioned, but you seem to be adamant to not make changes so I'll include this in my refactoring after this PR gets merged.